surface/secretsmanager/admin.py

```diff
@@ -0,0 +1,120 @@
+from django.contrib import admin
+from django.utils.html import format_html
+from django.urls import reverse
+from .models import Secret, SecretHistory, SecretLocation
+
+@admin.register(Secret)
+class SecretAdmin(admin.ModelAdmin):
+    list_display = (
+        'secret_preview',
+        'kind',
+        'source',
+        'git_source_link',
+        'location_links',
+        'status',
+        'criticality',
+        'locations_detail',
+        'verified',
+        'updated_at'
+    )
+    list_editable = ('status', 'criticality')
+    list_filter = ('source', 'status', 'criticality', 'verified', 'team', 'git_source', 'secret_hash')
+    search_fields = ('secret', 'source', 'kind', 'team', 'git_source__repo_url', 'secret_hash')
+    readonly_fields = ('created_at', 'updated_at', 'updated_by', 'version', 'secret_hash')
+    
+    def secret_preview(self, obj):
+        return obj.secret[:30] + '...' if len(obj.secret) > 30 else obj.secret
+    secret_preview.short_description = 'Secret'
+    
+    def git_source_link(self, obj):
+        if obj.git_source:
+            return format_html('<a href="{}" target="_blank">{}</a>', 
+                obj.git_source.repo_url, 
+                obj.git_source.repo_url)
+        return ''
+    git_source_link.short_description = 'Repository'
+    
+    def location_links(self, obj):
+        """Numbered links to specific files in repository"""
+        locations = obj.locations.all()
+        if locations:
+            links = []
+            max_links = 10
+            total_links = locations.count()
+
+            for idx, loc in enumerate(locations[:max_links], 1):
```

Comment:
Filter by the first 10 links and sort with creation date ascending (newer first). Better and lighter than loading everything and then doing the logic in Python.